packages/firebase_database/lib/src/firebase_database.dart

```diff
@@ -22,17 +23,73 @@ class FirebaseDatabase {
     });
   }
 
-  factory FirebaseDatabase() {
-    return new FirebaseDatabase.private(
-      const MethodChannel('plugins.flutter.io/firebase_database'),
-    );
-  }
+  static FirebaseDatabase _instance = new FirebaseDatabase._();
 
-  static FirebaseDatabase _instance = new FirebaseDatabase();
-
-  // Gets the instance of FirebaseDatabase for the default Firebase app.
+  /// Gets the instance of FirebaseDatabase for the default Firebase app.
   static FirebaseDatabase get instance => _instance;
 
   /// Gets a DatabaseReference for the root of your Firebase Database.
   DatabaseReference reference() => new DatabaseReference._(this, <String>[]);
+
+  /// The Firebase Database client will cache synchronized data and keep track
+  /// of all writes you’ve initiated while your application is running. It
+  /// seamlessly handles intermittent network connections and re-sends write
+  /// operations when the network connection is restored.
+  ///
+  /// However by default your write operations and cached data are only stored
+  /// in-memory and will be lost when your app restarts. By setting this value
+  /// to YES, the data will be persisted to on-device (disk) storage and will
+  /// thus be available again when the app is restarted (even when there is no
+  /// network connectivity at that time). Note that this property must be set
+  /// before creating your first Database reference and only needs to be called
+  /// once per application.
+  Future<Null> setPersistenceEnabled(bool enabled) {
+    return _channel.invokeMethod(
+      "FirebaseDatabase#setPersistenceEnabled",
+      { 'enabled': enabled },
```

Comment:
ok. I was optimizing for fewer lines of code but I see your point. I'll make the change.